### PR TITLE
Add new attributes to Vulnerability

### PR DESCRIPTION
**Pull request by danfaizer**
We need to enrich vulnerabilities with two new attributes in order to improve vulnerability filtering and classification.

`Category`: Defines if a finding is a vulnerability, a potential vulnerability (perhaps not exploitable), compliance or if the finding is just informational.

`AffectedResource`: Defines the exact resource where the finding has been detected.

Both attributes must be provided by the checks as is the component that have enough context to provide reliable data in these two new attributes.

This PR additionally includes testing and CI integration to run the tests.

### CODE REVIEW

**Review by ka3de — DISMISSED** [review]
LGTM.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+GOPATH=$(shell go env GOPATH)
+GOBIN=$(GOPATH)/bin
+GOSRC=$(GOPATH)/src
+
+$(GOBIN)/golangci-lint:
+	curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(GOBIN) v1.36.0
```
Comment (svalentino) [inline Makefile:6]: Even thought this is the "official" and documented way of installing `golangci-lint`, we should avoid curling and piping bash scripts in general: from a security perspective, it leaves us vulnerable to supply-chain attacks like [codecov](https://about.codecov.io/security-update/).

Easy fixes: 
- either perform integrity check on the downloaded script before executing it
- or use `go get github.com/golangci/golangci-lint/cmd/golangci-lint@<COMMIT HASH>`

Comment (danfaizer) [reply Makefile]: Good catch @svalentino 
I followed the same pattern (AKA copy/paste) other security tools were doing but indeed this bash-curl is a flaw path.
Fixed. 👍

Comment (svalentino) [reply Makefile]: Awesome 👍

**Review by ka3de — APPROVED** [review]
LGTM.

**Review by svalentino — APPROVED** [review]
LGTM

**Pull request merged**